PInfo[] parray
int16 best_particle
